Update Remove Listening History Button

mebarbosa opened the pull request:
## Description
- This removes the button instead of hide to avoid this issue https://github.com/Automattic/pocket-casts-android/pull/3453#issuecomment-2605485990

## Testing Instructions
- Smoke test the remove individual listening history
- ✅ Ensure this button displays only for when the user played the episode
- ✅ Ensure this button is not available for when the user favorited / downloaded an episode without playing it


## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-android/milestone/72">7.81 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

geekygecko commented:
Should the "Remove from history" button be on pages other than the "Listening History"? If I star an episode that's in my listening history and then open the starred page I can see the option when I multi-select.

|  |  |
| --- | --- |
| ![Screenshot_20250122_151434](https://github.com/user-attachments/assets/03e1a348-624a-4673-b971-ac0174b7c71f) | ![Screenshot_20250122_151450](https://github.com/user-attachments/assets/32c4c4ec-2703-44e5-9a10-79bca45ed5ec) |

mebarbosa commented:
> Should the "Remove from history" button be on pages other than the "Listening History"? If I star an episode that's in my listening history and then open the starred page I can see the option when I multi-select.
> 
@geekygecko I don't mind to change this, but I applied the same behavior that we implemented for the other buttons. For example, if I download an episode I can see the star option in downloads screen

geekygecko commented:
> I don't mind to change this, but I applied the same behavior that we implemented for the other buttons. For example, if I download an episode I can see the star option in downloads screen

It doesn't feel the same having the remove from history on the other screens. For example this is the Web Player, it only has the action on the listening history page. I think it might be best for us to ask the product team. 

<img width="1043" alt="Screenshot 2025-01-23 at 11 14 39 am" src="https://github.com/user-attachments/assets/bf91f07f-16d9-44b7-9987-e6dc3cfdcf59" />

<img width="995" alt="Screenshot 2025-01-23 at 11 14 48 am" src="https://github.com/user-attachments/assets/f1d0f3fc-fbae-4642-a9bb-542ca6381fcd" />

CookieyedCodes commented:
It could also be a swipe icon instead of archive just a proposal tho :)

![Screenshot_2025-01-23-10-22-49-74_28ad70af3b47247953fcd94176b9a9c1.jpg](https://github.com/user-attachments/assets/fd454fed-54a3-44b8-8b72-dc39747394ee)

mebarbosa commented:
> > I don't mind to change this, but I applied the same behavior that we implemented for the other buttons. For example, if I download an episode I can see the star option in downloads screen
> 
> It doesn't feel the same having the remove from history on the other screens. For example, this is the Web Player, it only has the action on the listening history page. I think it might be best for us to ask the product team. p1737593296509329-slack-C05RR9P9RAT
> 
Thanks for checking this with product team @geekygecko ! I have updated the code to remove this button from other screens

mebarbosa commented:
> It could also be a swipe icon instead of archive just a proposal tho :)
> 

Thanks for your suggestion @CookieyedCodes ! We are going to keep this implementation for now since our product team already suggested, but we will keep your feedback in mind

mebarbosa commented on modules/services/views/src/main/java/au/com/shiftyjelly/pocketcasts/views/multiselect/MultiSelectToolbar.kt:
```diff
@@ -56,7 +57,8 @@ class MultiSelectToolbar @JvmOverloads constructor(
                     item.setShowAsAction(MenuItem.SHOW_AS_ACTION_ALWAYS)
-                    item.isVisible = action.isVisible
+                    item.isVisible =
+                        if (action is MultiSelectEpisodeAction.RemoveListeningHistory && sourceView != SourceView.LISTENING_HISTORY) false else action.isVisible
```
I liked that!
